Include/internal/pycore_typeobject.h

```diff
@@ -269,6 +269,11 @@ extern unsigned int _PyType_GetVersionForCurrentState(PyTypeObject *tp);
 PyAPI_FUNC(void) _PyType_SetVersion(PyTypeObject *tp, unsigned int version);
 PyTypeObject *_PyType_LookupByVersion(unsigned int version);
 
+// Returns 0 on success or caller-specific error on failure.
+typedef int (*_py_validate_type)(PyTypeObject *);
+// Returns 0 on success, -1 if no type version could be assigned, or the error returned by validate
```

Comment:
Can you add a short high-level description of the function in addition to the return value documentation?